src/PhpPact/Standalone/PactConfig.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+namespace PhpPact\Standalone;
+
+use Composer\Semver\VersionParser;
+
+/**
+ * Class PactConfig.
+ */
+class PactConfig implements PactConfigInterface
+{
+    /**
+     * Consumer name.
+     *
+     * @var string
+     */
+    private string $consumer;
+
+    /**
+     * Provider name.
+     *
+     * @var string
+     */
+    private string $provider;
+
+    /**
+     * Directory to which the pacts will be written.
+     *
+     * @var null|string
+     */
+    private ?string $pactDir = null;
+
+    /**
+     * The pact specification version to use when writing the pact. Note that only versions 1, 2, 3 and 4 are currently supported.
+     *
+     * @var string
+     */
+    private string $pactSpecificationVersion;
+
+    /**
+     * File to which to log output.
+     *
+     * @var null|string
+     */
+    private ?string $log = null;
+
+    /**
+     * @var null|string
+     */
+    private ?string $logLevel = null;
+
+    /**
+     * `overwrite` or `merge`. Use `merge` when running multiple mock service
+     * instances in parallel for the same consumer/provider pair. Ensure the
+     * pact file is deleted before running tests when using this option so that
+     * interactions deleted from the code are not maintained in the file.
+     *
+     * @var string
+     */
+    private string $pactFileWriteMode = self::MODE_OVERWRITE;
```

Comment:
I think we need to change to:

```
private string $pactFileWriteMode = self::MODE_MERGE;
```

Because this pull request only allow only one interaction per consumer/provider pair (a pact json file).

But I'm not sure how to deal with this:

> Ensure the pact file is deleted before running tests when using this option

Reply:
Maybe updating `PactTestListener::startTestSuite` like this?

```php
class PactTestListener
{
    /**
     * @param TestSuite $suite
     *
     * @throws \Exception
     */
    public function startTestSuite(TestSuite $suite): void
    {
        $consumer = $this->mockServerConfig->getConsumer();
        $provider = $this->mockServerConfig->getProvider();
        $pactDir = $this->mockServerConfig->getPactDir();
        unlink($pactDir . \DIRECTORY_SEPARATOR . "{$consumer}-{$provider}.json");
    }
}
```

Reply:
IMO, save that for a separate PR after the FFI change.   If we do not support it today, let's keep it that way.

Reply:
I will note to create a separate PR later.

Reply:
See https://github.com/pact-foundation/pact-php/pull/293

Reply:
Resolve to make code reviewing easier. Feel free to reopen or add comment.